crates/bevy_reflect/src/type_registry.rs

```diff
@@ -18,6 +18,7 @@ use std::{any::TypeId, fmt::Debug, sync::Arc};
 /// [reflected]: crate
 /// [Registering]: TypeRegistry::register
 /// [crate-level documentation]: crate
+#[derive(Clone)]
```

Comment:
Can you help me understand how this would work with the implementation and with none TypeRegistryArc TypeRegistry's? I think your proposed solution would be essentially this as shown in the below example you did? My understanding of AsRef is that it is essentially a reference which the command wouldn't be able to save because the command can be longer lived than the reference. 

I see how we can just clone the TypeRegistryArc but in other scenarios like a custom TypeRegistry how would this work as a reference to the actual TypeRegistry wouldnt be valid.
```rust
fn insert_reflected<T: Resource + AsRef<TypeRegistry>(
  &mut self,
  component: Box<dyn Reflect>,
) -> &mut Self;

```

Reply:
Note that `bevy_reflect::TypeRegistry` is in reality `TypeRegistryArc`, so no
issues with cloning it.

I always thought it was confusing, and I think it should be changed, but that's
how it is today.

I'm with MrGVSV, it's a very bad idea to impl `Clone` on `TypeRegistry`, especially when we make so much effort to nudge users to use the `Arc` version instead, to avoid costly clones.

In any case you don't need to own it. The `TypeRegistry` would be extracted from the `world: &mut World` in the `Command::apply` implementation of the command.

The idea is as follow:

```rust
struct InsertReflectedWithRegistryCommand<T: Resource + AsRef<TypeRegistry>> {
    component: Box<dyn Reflect>,
    _t: PhantomData<fn(T)>,
}
impl<T: Resource + AsRef<TypeRegistry>> Command for InsertReflectedWithRegistryCommand<T> {
    fn apply(self, world: &mut World) {
        let registry = world.get_resource::<T>().unwrap().as_ref().clone();
        // do things with `registry`
    }
}
```

Reply:
This example really helped! Thank you and I was able to implement this and remove the clone derive for TypeRegistry